doc/developers/utilities.rst

```diff
@@ -43,6 +43,11 @@ should be used when applicable.
   be sliced or indexed using safe_index.  This is used to validate input for
   cross-validation.
 
+Some estimators are able to store intermediate results using ``Memory``
+instance. The func:`check_memory`: check that memory is a valid input.
```

Comment:
I would simplify the wording and replace your paragraph with something like:
```
The :func:`check_memory` checks that input is ``joblib.Memory``-like,
which means that it can be converted into a
``sklearn.externals.joblib.Memory`` instance (typically a str denoting
the ``cachedir``) or has the same interface.
```